lib/howitzer/mail_adapters/mailtrap.rb

```diff
@@ -0,0 +1,105 @@
+require 'howitzer/exceptions'
+require 'howitzer/mail_adapters/abstract'
+require 'howitzer/mailtrap_api/client'
+
+module Howitzer
+  module MailAdapters
+    # This class represents mailtrap mail adapter
+    class Mailtrap < Abstract
+      # Finds an email in storage
+      # @param recipient [String] an email
+      # @param subject [String]
+      # @param wait [Integer] how much time is required to wait an email
+      # @raise [EmailNotFoundError] if blank message
+
+      def self.find(recipient, subject, wait:)
+        message = {}
+        retryable(find_retry_params(wait)) { message = retrieve_message(recipient, subject) }
+        return new(message) if message.present?
+        raise Howitzer::EmailNotFoundError,
+              "Message with subject '#{subject}' for recipient '#{recipient}' was not found."
+      end
+
+      # @return [String] plain text body of the email message
+
+      def plain_text_body
+        message['text_body']
+      end
+
+      # @return [String] html body of the email message
+
+      def html_body
+        message['html_body']
+      end
+
+      # @return [String] stripped text
+
+      def text
+        message['text_body']
+      end
+
+      # @return [String] an email address specified in `From` field
+
+      def mail_from
+        message['from_email']
+      end
+
+      # @return [String] recipient emails separated with `, `
+
+      def recipients
+        message['to_email'].split ', '
+      end
+
+      # @return [String] when email was received
+
+      def received_time
+        Time.parse(message['created_at']).to_s
+      end
+
+      # @return [String] a real sender email address
+
+      def sender_email
+        message['from_email']
+      end
+
+      # @return [Array] attachments
+
+      def mime_part
+        retrieve_attachments(message)
+      end
+
+      # @raise [NoAttachmentsError] if no attachments present
+      # @return [Array] attachments
+
+      def mime_part!
+        files = mime_part
+        return files if files.present?
+        raise Howitzer::NoAttachmentsError, 'No attachments were found.'
+      end
+
+      def self.find_retry_params(wait)
+        {
+          timeout: wait || Howitzer.try(:mail_idle_timeout),
```

Comment:
try is not good here. Originally, if setting is missing but required, then we raise proper exception. If you introduce the option, then even if user will migrate existing project to last Howitzer version, he will be noticed about required missing setting